### PR TITLE
Hotels: Add stay 22 support for single hotel map screen

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js [hunk 1]
```diff
@@ -6,28 +6,45 @@ import { withMappedNavigationAndConfigProps } from 'react-navigation-props-mappe
 
 import type { AvailableHotelSearchInput } from '../../singleHotel/AvailableHotelSearchInput';
 import SingleHotelMap from '../../map/singleHotel/SingleHotelMapScreen';
```
Comment (RobinCsl) [inline app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js]:
```suggestion
import SingleHotelMapScreen from '../../map/singleHotel/SingleHotelMapScreen';
```

File: app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js [hunk 2]
```diff
@@ -6,28 +6,45 @@ import { withMappedNavigationAndConfigProps } from 'react-navigation-props-mappe
...
   currency: string,
 };
 
 class SingleHotelMapNavigationScreen extends React.Component<Props> {
-  render() {
+  renderInner = ({ apiProvider }) => {
+    const child =
+      apiProvider === 'booking' ? (
+        <React.Fragment>
+          <SingleHotelMap
```
Comment (RobinCsl) [inline app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js]:
```suggestion
          <SingleHotelMapScreen
```

File: app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js [hunk 3]
```diff
@@ -6,28 +6,45 @@ import { withMappedNavigationAndConfigProps } from 'react-navigation-props-mappe
...
+              hotelId: this.props.hotelId,
+              checkin: this.props.checkin,
+              checkout: this.props.checkout,
+              roomsConfiguration: this.props.roomsConfiguration,
+            }}
+            currency={this.props.currency}
+          />
+        </React.Fragment>
+      ) : (
+        <Stay22SingleHotelMapScreen />
```
Comment (RobinCsl) [inline app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js]: So booking.com hotel screen gets its props via navigation, but Stay22 does not need to pass down these props?

File: app/hotels/src/map/singleHotel/Stay22SingleHotelMapScreen.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            currency: $currency
+            checkin: $checkin
+            checkout: $checkout
+          ) {
+            ...SingleMap
+          }
+        }
+      `}
+      variables={{
+        checkin: sanitizeDate(checkin),
```
Comment (RobinCsl) [inline app/hotels/src/map/singleHotel/Stay22SingleHotelMapScreen.js]: Can these be obtained from the navigation like for booking.com screen?

File: app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js (hunk 3 above)
Comment (tbergq) [reply app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js]: If I would create the booking.com query today, I would probably just connect to the context as well. I could refactor it, so they both fetch from the context.

File: app/hotels/src/map/singleHotel/Stay22SingleHotelMapScreen.js (hunk 4 above)
Comment (tbergq) [reply app/hotels/src/map/singleHotel/Stay22SingleHotelMapScreen.js]: I think it is better to get it from context.

File: app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js (hunk 3 above)
Comment (tbergq) [reply app/hotels/src/navigation/singleHotel/SingleHotelMapNavigationScreen.js]: Refactored so none of them takes props.

**Comment by tbergq** [comment]
@RobinCsl 🚀

**Pull request merged**